content/documentation/pages/5-feature-guides/2-batch/3-partitioning.md

```diff
@@ -1,7 +1,260 @@
+                .setEnvironmentVariablesProvider(new SimpleEnvironmentVariablesProvider(this.environment));
+            partitionHandler.setMaxWorkers(1);
+            partitionHandler.setApplicationName("PartitionedBatchJobTask");
+
+            return partitionHandler;
+        }
+
+        @Bean // <2>
+        @Profile("!worker")
+        public Job partitionedJob(PartitionHandler partitionHandler) throws Exception {
+            Random random = new Random();
+            return this.jobBuilderFactory.get("partitionedJob" + random.nextInt())
+                .start(step1(partitionHandler))
+                .build();
+        }
+
+        @Bean // <3>
+        public Step step1(PartitionHandler partitionHandler) throws Exception {
+            return this.stepBuilderFactory.get("step1")
+                .partitioner(workerStep().getName(), partitioner())
+                .step(workerStep())
+                .partitionHandler(partitionHandler)
+                .build();
+        }
+
+        @Bean   // <4>
+        public Partitioner partitioner() {
+            return new Partitioner() {
+                @Override
+                public Map<String, ExecutionContext> partition(int gridSize) {
+
+                    Map<String, ExecutionContext> partitions = new HashMap<>(gridSize);
+
+                    for (int i = 0; i < GRID_SIZE; i++) {
+                        ExecutionContext context1 = new ExecutionContext();
+                        context1.put("partitionNumber", i);
+
+                        partitions.put("partition" + i, context1);
+                    }
+
+                    return partitions;
+                }
+            };
+        }
+
+        @Bean   // <5>
+        @Profile("worker")
+        public DeployerStepExecutionHandler stepExecutionHandler(JobExplorer jobExplorer) {
+            return new DeployerStepExecutionHandler(this.context, jobExplorer, this.jobRepository);
+        }
+
+        @Bean // <6>
+        public Step workerStep() {
+            return this.stepBuilderFactory.get("workerStep")
+                .tasklet(workerTasklet(null))
+                .build();
+        }
+
+        @Bean // <7>
+        @StepScope
+        public Tasklet workerTasklet(
+            final @Value("#{stepExecutionContext['partitionNumber']}") Integer partitionNumber) {
+
+            return new Tasklet() {
+                @Override
+                public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
+                    System.out.println("This tasklet ran partition: " + partitionNumber);
+
+                    return RepeatStatus.FINISHED;
+                }
+            };
+        }
+    }
+    ```
+
+    - <1> The [PartitionHandler](https://docs.spring.io/spring-batch/4.0.x/reference/html/index-single.html#partitionHandler) is the component that knows about the fabric of the remoting or grid environment. It sends StepExecution requests to the remote Steps.
```

Comment:
"fabric of the remoting or grid environment" <- it may not be clear to regular devs as to what we are trying to convey here